src/Howerpoint.hs

```diff
@@ -95,56 +124,48 @@ getTerminalSize = do
 
 loadPresentation :: String -> IO Presentation
 loadPresentation path = do
-  slides <- parseSlides path
+  slides' <- parseSlides path
   (w, h) <- getTerminalSize
-  ioRef <- newIORef $ PresentationState slides w (h - 3) 0
+  ioRef <- newIORef $ PresentationState slides' w (h - 3) 0
+
   displaySlide ioRef
   let
-    maxSlides = length slides
-    goToSlide i = moveSlide ioRef (\x -> if i > 0 && i <= maxSlides then i - 1 else x)
-    advanceSlides i = moveSlide ioRef (\x -> min (x + i) (maxSlides - 1))
-    goBackSlides i  = moveSlide ioRef (\x -> max (x - i) 0)
-    previousSlide = goBackSlides 1
-    nextSlide     = advanceSlides 1
+    g  = moveWith . const
+    nn = moveWith . (+)
+    pp = moveWith . subtract
+    p  = pp 1
+    n  = nn 1
+
+    moveWith f
+      = modifyIORef ioRef (moveSlide f) >> displaySlide ioRef
 
     resetSize = do
-      (w, h) <- getTerminalSize
-      modifyIORef ioRef (\p -> p {width = w, height = h - 3})
-      displaySlide ioRef
+      (w', h') <- getTerminalSize
+      setSize w' (h' - 3)
 
-    setSize w h = do
-      modifyIORef ioRef (\p -> p {width = w, height = h})
+    setSize w' h' = do
+      modifyIORef ioRef (\p' -> p' {width = w', height = h'})
       displaySlide ioRef
 
     showSize = do
       ps <- readIORef ioRef
-      putStrLn $ (show $ width ps) ++ "x" ++ (show $ height ps)
+      putStrLn $ show (width ps) ++ "x" ++ show (height ps)
 
     codeFromSlide = filterCode . getCurrentSlide <$> readIORef ioRef
-  return $ Presentation {
-    n = nextSlide,
-    p = previousSlide,
-    nn = advanceSlides,
-    pp = goBackSlides,
-    g = goToSlide,
-    resetSize = resetSize,
-    setSize = setSize,
-    showSize = showSize,
-    codeFromSlide = codeFromSlide
-  }
+  return Presentation {..}
```

Comment:
I've matched the names of the functions in the 'let' block with the `Presentation` fields' names, so the `RecordWildCards` extension can pack them in for us from the local scope.